src/jvm_wrapper/memory/memory_manager.cpp

```diff
@@ -1,27 +1,19 @@
 #include "memory_manager.h"
 
 #include "binding/kotlin_binding_manager.h"
+#include "script/jvm_script_manager.h"
+#include "shared_buffer.h"
+#include "transfer_context.h"
+#include "type_manager.h"
 
 bool MemoryManager::check_instance(JNIEnv* p_raw_env, jobject p_instance, jlong p_raw_ptr, jlong instance_id) {
     auto* instance {reinterpret_cast<Object*>(static_cast<uintptr_t>(p_raw_ptr))};
     return instance == ObjectDB::get_instance(static_cast<ObjectID>(static_cast<uint64_t>(instance_id)));
 }
 
-void MemoryManager::bind_instance(JNIEnv* p_raw_env, jobject p_instance, jlong instance_id, jobject p_object) {
-    ObjectID id {static_cast<uint64_t>(instance_id)};
-    jni::JObject j_object {p_object};
-    KotlinBindingManager::bind_object(id, j_object);
-}
-
-void MemoryManager::unbind_instance(JNIEnv* p_raw_env, jobject p_instance, jlong instance_id) {
-    ObjectID id {static_cast<uint64_t>(instance_id)};
-    KotlinBindingManager::unbind_object(id);
-}
-
 void MemoryManager::decrement_ref_counter(JNIEnv* p_raw_env, jobject p_instance, jlong instance_id) {
     Object* obj = ObjectDB::get_instance(static_cast<ObjectID>(static_cast<uint64_t>(instance_id)));
```

Comment:
Simplification:
```cpp
if (Object* obj { ObjectDB::get_instance(static_cast<ObjectID>(static_cast<uint64_t>(instance_id))) }) { 
    KotlinBindingManager::decrement_counter(reinterpret_cast<RefCounted*>(obj));
}
```

Reply:
It's a mistake, I added the if during debug but I don't even need it. I know the RefCounted can't be dead before I decrement it.